core/src/main/java/com/microsoft/applicationinsights/telemetry/BaseSampleSourceTelemetry.java

```diff
@@ -17,6 +17,8 @@ public void reset() {
     @Override
     protected void setSampleRate(Envelope envelope) {
         Double currentSP = getSamplingPercentage();
-        envelope.setSampleRate(currentSP);
+        if (currentSP != null) {
```

Comment:
@SergeyKanzhelev getSamolingPercentage() returns null and null gets assigned to currentSP. Then the call to envelope.setSampleRate(currentSP) sets null to the sampleRate of the envelope overriding its default value of 100.00D. That was the flow before this change and therefore I added the null check to see if the currentSP has null value then don't override the default.